backend/inspirehep/pidstore/models.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import datetime
+
+import structlog
+from invenio_db import db
+from invenio_pidstore.errors import PIDInvalidAction
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from sqlalchemy.exc import IntegrityError, SQLAlchemyError
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy_utils.models import Timestamp
+
+from inspirehep.pidstore.errors import (
+    PidRedirectionMissing,
+    WrongPidTypeRedirection,
+    WrongRedirectionPidStatus,
+)
+
+LOGGER = structlog.getLogger()
+
+
+class InspireRedirect(db.Model, Timestamp):
+    __tablename__ = "inspire_pidstore_redirect"
+    __table_args__ = (
+        db.Index(
+            "ix_inspire_pidstore_redirect_new_record", "new_pid_type", "new_pid_value"
+        ),
+    )
+
+    original_pid_type = db.Column(db.String(6), nullable=False, primary_key=True)
+    original_pid_value = db.Column(db.String(255), nullable=False, primary_key=True)
+
+    new_pid_type = db.Column(db.String(6), nullable=False)
+    new_pid_value = db.Column(db.String(255), nullable=False)
```

Comment:
why not make these foreign keys on the pidstore instead? you'd always need to join on the pidstore to get the pid metadata, but you need to do it anyway if you want to get the `object_uuid`.

Reply:
:+1: